src/macro-core/macro-condition-file.cpp

```diff
@@ -205,39 +242,79 @@ std::string MacroConditionFile::GetShortDesc() const
 
 static void populateFileTypes(QComboBox *list)
 {
-	list->addItem(obs_module_text("AdvSceneSwitcher.condition.file.local"));
-	list->addItem(
-		obs_module_text("AdvSceneSwitcher.condition.file.remote"));
+	static constexpr std::array<
+		std::tuple<MacroConditionFile::FileType, std::string_view>, 2>
+		fileTypes = {{{MacroConditionFile::FileType::LOCAL,
+			       "AdvSceneSwitcher.condition.file.local"},
+			      {MacroConditionFile::FileType::REMOTE,
+			       "AdvSceneSwitcher.condition.file.remote"}}};
+
+	for (const auto &[type, name] : fileTypes) {
+		list->addItem(obs_module_text(name.data()),
+			      static_cast<int>(type));
+	}
 }
 
 static void populateConditions(QComboBox *list)
 {
-	list->addItem(
-		obs_module_text("AdvSceneSwitcher.condition.file.type.match"));
-	list->addItem(obs_module_text(
-		"AdvSceneSwitcher.condition.file.type.contentChange"));
-	list->addItem(obs_module_text(
-		"AdvSceneSwitcher.condition.file.type.dateChange"));
+	static constexpr std::array<
+		std::tuple<MacroConditionFile::ConditionType, std::string_view>,
```

Comment:
Just curious - wh this instead of `map`

Reply:
Ah just some unnecessary constexpr optimization ... no reason in particular.